lib/active_model/serializer.rb

```diff
@@ -164,6 +164,14 @@ def json_key
       end
     end
 
+    def id
+      object.id if object
+    end
+
+    def type
+      object.class.to_s.demodulize.underscore.pluralize
```

Comment:
Yes, easily overriding the type is one of the main reasons I put this here. One can create an `ApplicationSerializer` and use different inflection rules.

I put the id here as well because it made thing more symmetrical, but I can remove it.